src/main/java/com/deque/axe/android/rules/hierarchy/base/TouchSize.java

```diff
@@ -43,17 +43,32 @@ public String runRule(AxeProps axeProps) {
     final long height = frame.height();
     final long width = frame.width();
 
-    if (dpi <= 0 || height < 0 || width < 0) {
+    final int screenHeight = axeProps.get(AxeProps.Name.SCREEN_HEIGHT, Integer.class);
+    final int screenWidth = axeProps.get(AxeProps.Name.SCREEN_WIDTH, Integer.class);
+
+    if (dpi <= 0 || height < 0 || width < 0
+            || touchTargetOffscreen(frame, screenHeight, screenWidth)) {
       return AxeStatus.INCOMPLETE;
     }
 
     final long adjustedHeight = Math.round(height / dpi);
     final long adjustedWidth = Math.round(width / dpi);
 
+
     if (adjustedHeight < expectedSize || adjustedWidth < expectedSize) {
       return AxeStatus.FAIL;
     } else {
       return AxeStatus.PASS;
     }
   }
+
+  private boolean touchTargetOffscreen(AxeRect frame, int screenHeight, int screenWidth) {
```

Comment:
Should be an internal, utitlity function on AxeView

`isOffScreen(screenHeight, screenWidth)`